app/controller.go

```diff
@@ -133,3 +140,86 @@ func startController(c *cli.Context) error {
 	})
 	return http.ListenAndServe(controlListener, router)
 }
+
+func checkPrerequisits(c *controller.Controller, replicas []types.Replica) error {
+	if c.IsSnapDeletionInProgress {
+		return fmt.Errorf("Snapshot deletion process is in progress, %s is getting deleted", c.SnapshotName)
+	}
+
+	rf := c.ReplicationFactor
+	if len(replicas) != rf {
+		return fmt.Errorf("Can not remove a snapshot because, RF: %v, replica count: %v", rf, len(replicas))
+	}
+
+	for _, rep := range replicas {
+		r := rep // pin it
+		if err := c.IsReplicaRW(&r); err != nil {
```

Comment:
do we need this? however, replica errors out snap delete if its not in RW, right?

Reply:
doing this here will hold the lock for longer time

Reply:
yes, some replicas may be in WO mode.